core/libp2p/transport/tcp/tcp_util.hpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TCP_CONNECTION_UTIL_HPP
+#define KAGOME_TCP_CONNECTION_UTIL_HPP
+
+#include <sstream>
+#include <system_error>  // for std::errc
+
+#include <boost/asio.hpp>
+#include <boost/lexical_cast.hpp>
+#include <gsl/span>
+#include <outcome/outcome.hpp>
+#include "libp2p/multi/multiaddress.hpp"
+
+namespace libp2p::transport::detail {
+  template <typename T>
+  inline outcome::result<multi::Multiaddress> make_endpoint(T &&endpoint) {
```

Comment:
what are the constraints for the type T here? pls, leave a comment

Reply:
it is internal method (see `namespace detail `)

Reply:
Anyway, would be great to have short description of type T. 
Also, please make the function name camelCase